### PR TITLE
Add linting / static analysis tools

### PR DESCRIPTION
**Pull request by duckth**
This pull requests adds various linters / static analyzers to our different tools / languages, and sets up a [Lefthook](https://github.com/evilmartians/lefthook) config to run them as pre-push git hooks.
**Ruby**: I've chosen Rubocop over standardRB because it seems like most of us had problems getting it to run, with tools like Solargraph and RubyLSP using Rubocop for diagnostics. So, to spare us the trouble of managing multiple configs and tweaking editor configs, let's just go with Rubocop (even though Standard is faster and has "saner" defaults IMHO). Specifically, I've chosen the [Shopify](https://en.wikipedia.org/wiki/Shopify)-config for rubocop, since it has some much saner settings than the default rubocop ones.
**GitHub Actions**: Uses actionlint to lint the files when edits are made in .github/*
**Vagrant**: Runs the built in `vagrant validate` command.
**Docker**: Uses Hadolint.

Also fixes various issues reported by these new linters.

### CODE REVIEW

File: myapp.rb [hunk 1]
```diff
@@ -89,71 +89,71 @@ def current_user
...
 
-  'You were logged out'
+  "You were logged out"
 end
 
-post '/add_message' do
+post "/add_message" do
   if !session[:user_id]
-    return status 401
+    return status(401)
```
Comment (A-Guldborg) [inline myapp.rb:140]: Is `status` not ignored alongside `body` and `header` in the `.rubocop.yml`? Why are parentheses needed here?

File: myapp.rb [hunk 2]
```diff
@@ -165,20 +165,20 @@ def current_user
   erb :timeline, layout: :layout
 end
 
-get '/:username/follow' do
-  return status 401 unless logged_in?
+get "/:username/follow" do
+  return status(401) unless logged_in?
```
Comment (A-Guldborg) [inline myapp.rb:169]: Same as above

File: myapp.rb [hunk 3]
```diff
@@ -165,20 +165,20 @@ def current_user
...
 
   whom = User.find_by_username(params[:username])
 
-  return status 404 if whom.nil?
+  return status(404) if whom.nil?
```
Comment (A-Guldborg) [inline myapp.rb:173]: Same as above

File: myapp.rb [hunk 4]
```diff
@@ -165,20 +165,20 @@ def current_user
...
 
   current_user.following << whom
   flash[:success] = "You are now following &#34;#{params[:username]}&#34;"
   redirect("/#{params[:username]}")
 end
 
-get '/:username/unfollow' do
-  return status 401 unless logged_in?
+get "/:username/unfollow" do
+  return status(401) unless logged_in?
```
Comment (A-Guldborg) [inline myapp.rb:181]: Same as above

File: simapi/sim_api.rb [hunk 5]
```diff
@@ -162,20 +165,20 @@ def update_latest(request)
...
     to_follow = User.find_by_username(request_data[:follow])
-    return [400, 'User to follow not found'] if to_follow.nil?
+    return [400, "User to follow not found"] if to_follow.nil?
 
     user.following.append(to_follow)
     status 204
   elsif request_data.key?(:unfollow)
-    to_unfollow = User.find_by_username(request_data[:unfollow])
-    return [400, 'User to unfollow not found'] if to_unfollow.nil?
+    to_unfollow = user.following.find_by_username(request_data[:unfollow])
```
Comment (A-Guldborg) [inline simapi/sim_api.rb:180]: Should this be part of this PR?

**Review by A-Guldborg — CHANGES_REQUESTED** [review]

File: simapi/sim_api.rb (hunk 5 above)
Comment (duckth) [reply simapi/sim_api.rb:180]: It should not

**Review by A-Guldborg — APPROVED** [review]
LGTM 🚀

**Review by fredpetersen — APPROVED** [review]

**Pull request merged**